openapi.yaml

```diff
@@ -25,254 +37,677 @@ security:
+          type: string
+          description: "ID of a pizza to retrive"
+      responses:
+        200:
+          description: "Returns true is the resource was deleted"
           schema:
-            $ref: '#/definitions/ErrorResult'
-        '500':
-          $ref: '#/responses/500'
-  /pets/{id}:
+            type: boolean
+            example: true
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+
+  /pizzas/{pizzaID}/ingredients:
     get:
-      summary: Info for a specific pet
+      description: "Get the ingredients from the pizza with ID pizzaID"
       tags:
-        - pets
-      description: Get pet by unique ID
-      operationId: getPetById
+        - pizza
+        - ingredient
       parameters:
-        - name: id
-          in: path
-          description: Unique Pet ID, UUID format
+        - in: path
+          name: pizzaID
           required: true
           type: string
+          description: "ID of a pizza to retrive"
       responses:
-        '200':
-          description: Successful response
+        200:
+          description: "A list of ingredients"
           schema:
-            $ref: '#/definitions/PetResult'
-        '404':
-          description: PetResource not found
+            $ref: "#/definitions/IngredientsResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /pizzas/{pizzaID}/ingredients/{ingredientID}:
+    patch:
+      description: add the ingredient with ID ingredientID to the pizza with ID pizzaID
+      tags:
+        - pizza
+        - ingredient
+      parameters:
+        - in: path
+          name: ingredientID
+          required: true
+          type: string
+          description: "ID of the ingredient to add"
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: "ID of a pizza to update"
+      responses:
+        200:
+          description: "Returns true is the resource was updated"
           schema:
-            $ref: '#/definitions/ErrorResult'
-        '500':
-          $ref: '#/responses/500'
-parameters:
-  pageNumber:
-    name: page[number]
-    in: query
-    type: integer
-    required: false
-    description: Page number of results
-    default: 1
-    minimum: 1
-  pageSize:
-    name: page[size]
-    in: query
-    type: integer
-    required: false
-    description: Number of results to return
-    default: 25
-    maximum: 500
-    minimum: 1
-responses:
-  '400':
-     description: Bad request
-     schema:
-       $ref: '#/definitions/ErrorResult'
-  '500':
-     description: Internal server error
-     schema:
-       $ref: '#/definitions/ErrorResult'
-securityDefinitions:
-  OAuth2:
-    type: oauth2
-    flow: application
-    tokenUrl: https://api.oregonstate.edu/oauth2/token
-    scopes:
-      full: Full access to the API
+            type: boolean
+            example: true
```

Comment:
We can't just return a boolean as a result. It must be a JSON object.

Ref: https://jsonapi.org/format/#crud-updating
> The PATCH request MUST include a single resource object as primary data. The resource object MUST contain type and id members.

Reply:
If I'm reading [this](https://jsonapi.org/format/#crud-updating-responses-204) correctly, an update with no side effects can just return a `204` right?

Reply:
Actually `200` and `204` are both acceptable:

> If an update is successful and the server doesn’t update any attributes besides those provided, the server MUST return either a 200 OK status code and response document (as described above) or a 204 No Content status code with no response document.

We usually return a `200` if there's no sensitive data returned. Example: https://github.com/osu-mist/notes-api/blob/master/openapi.yaml#L167-L170
Just curious, why you choose `204` instead of `200` here?

Reply:
204 seemed simpler since there's no body to return. 200 makes sense if that's the standard for the team though.